### ISSUE TITLE
New Rule Proposal: Enforce indentation of ternary expressions on multiple lines

### ISSUE THREAD

**Issue by kemitchell** [eslint/eslint#7420]
**Please describe what the rule should do:**

Enforce predictable indentation of ternary expression alternatives on separate lines.

``` javascript
var location = env.development
  ? 'localhost'
  : 'www.api.com'
```

``` javascript
var location = env.development
    ? 'localhost'
    : 'www.api.com'
```

Neither [multiline-ternary](http://eslint.org/docs/rules/multiline-ternary) nor [operator-linebreak](http://eslint.org/docs/rules/operator-linebreak) enforce such indentation.

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Less than expected indentation:

``` javascript
var location = env.development
? 'localhost'
: 'www.api.com'
```

Irregular indentation:

``` javascript
var location = env.development
   ? 'localhost'
       : 'www.api.com'
```

More than expected indentation:

``` javascript
var location = env.development
              ? 'localhost'
              : 'www.api.com'
```

**Why should this rule be included in ESLint (instead of a plugin)?**

The rule would fill a gap in core's enforcement of fixed and predictable indentation.

---

Related: feross/eslint-config-standard#46

cc @feross @dcousens

**Comment by not-an-aardvark** [comment]
Thanks for the suggestion. I think we should check for this, but in my opinion it would be better if the existing [`indent`](http://eslint.org/docs/rules/indent) rule was in charge of it (either by default or with an opt-in option), rather than having a separate rule just for ternary expressions. In my opinion, not checking ternary expressions is a bug in the indent rule.

(For what it's worth, there's a [rewrite](https://github.com/not-an-aardvark/eslint/tree/indent-rewrite) of the indent rule in progress, which checks ternary expressions by default along with some other cases that aren't currently checked.

**Comment by platinumazure** [comment]
Agreed with @not-an-aardvark, this absolutely belongs in "indent".

**Comment by nebirhos** [comment]
Hi @not-an-aardvark!

I'm using v4.15.0 and looks like the indentation is still not enforced for multiline expressions:

```
let valid = 1 +
2 +
     3 +
4 +
         5;
```

Did you decide to not implement this feature in the rule rewrite?

**Comment by not-an-aardvark** [comment]
This issue is specifically about ternary expressions. The `indent` rule currently ignores binary expressions that span multiple lines, although hopefully this will change in the future (see https://github.com/eslint/eslint/issues/8978)

**Comment by nebirhos** [comment]
@not-an-aardvark thanks for the clarification! I'll follow the updates on the issue then.
